examples/force-an-org-custom-flow-vanilla-js/src/main.js

```diff
@@ -0,0 +1,144 @@
+      e.preventDefault();
+
+      const formData = new FormData(e.target);
+      const emailAddress = formData.get("email");
+      const password = formData.get("password");
+
+      try {
+        // Start the sign-up process using the email and password provided
+        await clerk.client.signUp.create({ emailAddress, password });
+        await clerk.client.signUp.prepareEmailAddressVerification();
+        // Hide first-factor form
+        document.getElementById("sign-up").setAttribute("hidden", "");
+        document.getElementById("sign-in").setAttribute("hidden", "");
+
+        // Show verification form
+        document.getElementById("verifying").removeAttribute("hidden");
+      } catch (error) {
+        // See https://clerk.com/docs/custom-flows/error-handling
+        // for more info on error handling
+        console.error(error);
+      }
+    });
+
+  // Handle the verification form
+  document.getElementById("verifying").addEventListener("submit", async (e) => {
+    const formData = new FormData(e.target);
+    const code = formData.get("code");
+
+    try {
+      // Use the code the user provided to attempt verification
+      const signUpAttempt =
+        await clerk.client.signUp.attemptEmailAddressVerification({
+          code,
+        });
+
+      // Now that the user is created, set the session to active.
+      // TODO: Update this with clerk.setCurrentSession(signUpAttempt.createdSessionId)
+      await clerk.setActive({ session: signUpAttempt.createdSessionId });
+      // Display organization selection UI if session doesn't have a organization selected
+      await displayNextTaskIfAny();
+    } catch (error) {
+      // See https://clerk.com/docs/custom-flows/error-handling
+      // for more info on error handling
+      console.error(error);
+    }
+  });
+
+  // Handle organization selection
+  document
+    .getElementById("create-organization")
+    .addEventListener("submit", async (e) => {
+      e.preventDefault();
+
+      const inputEl = document.getElementById("name");
+
+      if (!inputEl) {
+        return;
+      }
+
+      clerk
+        .createOrganization({ name: inputEl.value })
+        .then((res) => console.log(res))
+        .catch((error) => console.log("An error occurred:", error));
+
+      // Hide the organization creation form after successful creation
+      document
+        .getElementById("create-organization")
+        .setAttribute("hidden", "true");
+      // Handle any remaining tasks in the flow
+      await displayNextTaskIfAny();
+    });
+}
+
+// Check for any pending tasks (like organization creation) and display the corresponding UI element
+async function displayNextTaskIfAny() {
+  const task = clerk.currentTask;
+
+  const mapTaskKeyToElementId = {
+    org: "create-organization",
+  };
+
+  // Here we're toggling UI visibility, but you could also navigate to different pages
+  // with `Clerk.__experimental_nextTask({ redirectUrlComplete })`
```

Comment:
Considering deprecating this one in favor of `navigateToTask` 

The main goal of this method is to abstract the logic of navigation from a `pending` to an `active` transition. The SDK has to deal with a lot of internal complexity on behalf of the developer, such as setting a transitive state for accessors and doing the actual status transition check.